Update tests for OSX

eam opened the pull request.

yahoocla commented:
CLA is valid!

areese commented on src/test/func-test/test_misc_func.cc:
```diff
@@ -143,9 +143,14 @@ class TestMiscFunc : public TestMiscFuncSuit {
+#endif
   CPPUNIT_TEST(test_misc_func_35);
+#ifdef __linux__
   CPPUNIT_TEST(test_misc_func_36);
```
can you add a comment for what this test is that's not supported on non-linux?
Is it also windowed mode?

areese commented on src/test/pool_test/test_ccmp.cc:
```diff
@@ -26,10 +26,13 @@ class YccmpTest: public CppUnit::TestFixture {
   CPPUNIT_TEST_SUITE(YccmpTest);
    
+#ifdef __linux__
+// uses /proc, does not exist on OSX
```
Shouldn't this be #ifndef _DARWIN_ or something.
There should be some sort of /proc on the later bsd's.  (I don't have a later bsd handy(

areese commented:
looks good to me.
It'd be nice to treat it as 3 OSes instead of just linux vs OS-X.

eam commented:
There are a few tests that absolutely are linux-only; I figured I'd err on the side of disabling with the assumption that these have mostly been developed for linux. Don't have a FreeBSD box handy to test on, unfortunately.

I'll switch it around to darwin-only excludes.

timrc-git commented:
Several thoughts:
a) this pretty much clinches that I should add a feature flag to disable windowed-mode
b) my apologies for not disabling the robust-lock tests on that feature flag
c) if possible, I'd like to pull the makefile differences into Makefile.base
    Or just unconditionally set DYLD_LIBRARY_PATH, or use RPATH/RUNPATH

I'll take a crack at this later (need to get some other stuff done first).

Thanks!
.timrc

eam commented:
Oh, of course. I should've wrapped those in `HAVE_ROBUST_PTHREADS`.

eam commented:
I moved the LD_LIB_PATH bits into `Makefile.base`, commented the reasons for excluding tests and switched to using `HAVE_ROBUST_PTHREADS`. I think this is ready to go.

timrc-git commented:
Thanks Evan!

After I merge this, I plan to add a flag for windowed mode. 
Hopefully, I can twist your arm to retest then.

eam commented:
Absolutely.
